docs/get-started/getting-started-kubernetes.md

```diff
@@ -0,0 +1,256 @@
+  [GPU discovery script](getting-started-on-prem.md#install-the-gpu-discovery-script).
+  
+   Put the 2 jars -- `rapids-4-spark_<version>.jar`, `cudf-<version>.jar`  and `getGpusResources.sh` in the same directory as `spark`.
+
+3. Download the sample [Dockerfile.cuda](Dockerfile.cuda) in the same directory as `spark`.
+
+   Basically the sample Dockerfile.cuda will copy the `spark` directory's several sub-directories into `/opt/spark/`, 
+   and copy above 2 jars and `getGpusResources.sh` into `/opt/sparkRapidsPlugin` inside the Docker image.
+   
+   So please look into the Dockerfile.cuda to make sure the file names are correct, and modify if needed.
+   
+   Currently the directory in the local machine should look as below:
+   ```shell 
+   $ ls
+   Dockerfile.cuda   cudf-<version>.jar   getGpusResources.sh   rapids-4-spark_<version>.jar   spark
+   ```
+
+4. Build the Docker image with proper repository name and tag and push to repository
+   ```shell 
+   export IMAGE_NAME=xxx/yyy:tag
+   docker build . -f Dockerfile.cuda -t $IMAGE_NAME
+   docker push $IMAGE_NAME
+   ```
+
+## Run Spark Applications in Kubernetes Cluster
+
+### spark-submit example pi job in `cluster` mode.
+
+This is a good way to test if RAPIDS plugin can be found.
+Because `ClassNotFoundException` is a common error if the Driver can not 
+find the jar for `com.nvidia.spark.SQLPlugin`:
+```
+Exception in thread "main" java.lang.ClassNotFoundException: com.nvidia.spark.SQLPlugin
+```
+
+Here is an example job: 
+
+```shell
+export SPARK_HOME=~/spark
+export IMAGE_NAME=xxx/yyy:tag
+export K8SMASTER=k8s://https://<k8s-apiserver-host>:<k8s-apiserver-port>
+export SPARK_NAMESPACE=default
+export SPARK_DRIVER_NAME=exampledriver
+
+$SPARK_HOME/bin/spark-submit \
+     --master $K8SMASTER \
+     --deploy-mode cluster  \
+     --name examplejob \
+     --class org.apache.spark.examples.SparkPi \
+     --conf spark.rapids.sql.concurrentGpuTasks=1 \
+     --conf spark.executor.instances=1 \
+     --conf spark.executor.resource.gpu.amount=1 \
+     --conf spark.executor.memory=4G \
+     --conf spark.executor.cores=1 \
+     --conf spark.task.cpus=1 \
+     --conf spark.task.resource.gpu.amount=1 \
+     --conf spark.rapids.memory.pinnedPool.size=2G \
+     --conf spark.locality.wait=0s \
+     --conf spark.sql.files.maxPartitionBytes=512m \
+     --conf spark.sql.shuffle.partitions=10 \
+     --conf spark.plugins=com.nvidia.spark.SQLPlugin \
+     --conf spark.kubernetes.namespace=$SPARK_NAMESPACE  \
+     --conf spark.kubernetes.driver.pod.name=$SPARK_DRIVER_NAME  \
+     --conf spark.executor.resource.gpu.discoveryScript=/opt/sparkRapidsPlugin/getGpusResources.sh \
+     --conf spark.executor.resource.gpu.vendor=nvidia.com \
+     --conf spark.kubernetes.container.image=$IMAGE_NAME \
+     --conf spark.executor.extraClassPath=/opt/sparkRapidsPlugin/rapids-4-spark_<version>.jar:/opt/sparkRapidsPlugin/cudf-<version>.jar   \
```

Comment:
extraClassPath shouldn't be needed with kubernetes

Reply:
@tgravescs This depends on: When building the container image, do we put the 2 jars into "jars" subdirectory of spark or not in the Docker container.  If yes, then no need, because those 2 jars are already there. However if we just copy it into /opt/sparkRapidsPlugin in Docker container, then we need to specify it here. This is to give a better clear picture for customers imo.

Reply:
I yes there are multiple ways to do this, this doc is putting them in /jars do not needed. I'm fine with adding something else saying if you don't put them in /jars then you may need to include them somehow, or we document all the known ways of doing this.  For instance you could also use --jars with spark -submit if you have setup shared storage space. I honestly don't know if you do this if they need to be included in extra class path.

Reply:
If using --jars, then no need since those jars after --jars will be sent to each executors.

Reply:
so it will depends on what spark does with them and who downloads them. If the jars aren't downloaded before starting the executor and put in class path then we still may need the extraClassPath options

Reply:
I haven't ever tried this or looked at the k8s code for this case.

Reply:
May need to use spark.kubernetes.file.upload.path to point to a shared storage.

Reply:
ming this is going by these instructions, the jars are in the spark/jars directory so we should remove them from this sample command and add in a separately section if you want to states other ways to do it.

Reply:
I just added some words to explain that:
`Note: If here you decide to put above 2 jars in the spark/jars directory which will be copied into /opt/spark/jars directory in Docker image, then in the future you do not need to specify spark.driver.extraClassPath or spark.executor.extraClassPath using cluster mode. This example just shows you a way to put customized jars or 3rd party jars.`

See if it looks good to you. Feel free to reword if needed.